lib/video_merger.ex

```diff
@@ -62,21 +62,21 @@ defmodule Membrane.VideoMerger do
 
     state
     |> BufferQueue.enqueue_list(id, buffers)
-    |> get_actions(redemand: :output)
+    |> get_actions()
   end
 
-  defp get_actions(state, fallback) do
+  defp get_actions(state, pad_ref \\ nil) do
     case BufferQueue.dequeue_buffers(state) do
-      {:ok, :end_of_stream} ->
-        {{:ok, fallback ++ [end_of_stream: :output]}, %{}}
+      {:empty, [], new_state} ->
+        {{:ok, [notify: pad_ref, end_of_stream: :output]}, new_state}
```

Comment:
`new_state` is now always returned, so you can extract that and in `case` only match `{atom, buffers}`, return actions and after `case` return `{{:ok, actions}, new_state}`